mesa: update to 25.2.5

MingcongBai opened the pull request:
Topic Description
-----------------
- fastfetch: rebuild to enable chafa and directx-headers
- mesa+32: update to 25.2.5+dxheaders1.618.2
    - We cannot use system directx-headers because the system one ships two
      64-bit static libraries.
    - Drop the already upstreamed patch for graphical corruption on Intel
      cards.
    - Track patches at AOSC-Tracking/mesa @ aosc/mesa-25.2.5
      (HEAD: ff4dcf999bdd04254add94999c64a0d359d45de9).
- mesa: update to 25.2.5
    - Reduce the feature set of stage 2 to simplify the logic.
    - Remove or modify deprecated or removed meson flags in stage 2 to fix meson
      setup failure.
    - Build with system directx-headers.
    - Drop the already upstreamed patch for graphical corruption on Intel
      cards.
    - Track patches at AOSC-Tracking/mesa @ aosc/mesa-25.2.5
      (HEAD: ff4dcf999bdd04254add94999c64a0d359d45de9).
- directx-headers: New, 1.618.2
  Separated from Mesa so it can be used by other packages.
- mesa+32: update to 25.2.4+dxheaders1.816.1
    - Enable all AArch64-specific drivers on amd64, helping with AArch64-x86
    emulation with sysroot-driven solutions \(such as FEX\).
    - Disable Rust features due to 32-bit rustc 1.90.0 brokenness.
    - Backport an upstream fix for graphical corruption on Intel cards.
    - Track patches at AOSC-Tracking/mesa @ aosc/25.2.4
    \(HEAD: b96ef130727285f6d1adcb1154468d46c46f8e0a\).
- rustc+32: update to 1.90.0
- mesa: update to 25.2.4+dxheaders1.618.0
    - Enable all AArch64-specific drivers on amd64, helping with AArch64-x86
    emulation with sysroot-driven solutions \(such as FEX\).
    - \(loongson3\) Disable rust features \(rusticl, gfxstream Vulkan\) due to
    libc \(crate\) brokenness.
    - Restore DirectX-Headers.pc.
    - Track patches at AOSC-Tracking/mesa @ aosc/25.2.4
    \(HEAD: 31b4a6f76b163291caa5dd9c62fe9178fb7d9f05\).
- mesa: update to 25.2.3+dxheaders1.618.1
    Track patches at AOSC-Tracking/mesa @ aosc/mesa-25.2.3
    \(HEAD: 4c9d45df71549deffc35e1c8c388c986bdec8ca7\).
- mesa: update to 25.2.2+dxheaders1.616.0
    - Backport a patch for Intel ANV \(Vulkan\) on non-4KiB kernel pages.
    - Track patches at AOSC-Tracking/mesa @ aosc/mesa-25.2.2
    \(HEAD: 59d6468b58382950b6aec8fb4dcf45c903eea4ab\).
- mesa: update to 25.2.1+dxheaders1.616.0
    Track patches at AOSC-Tracking/mesa @ aosc/mesa-25.2.1
    \(HEAD: 4c184e7a1fc82103a07e1e5e63fe74a15f582b25\).
- mesa: update to 25.2.0+dxheaders1.616.0
    - Revisit meson.options and enable options explicitly.
    - Revise MESON_AFTER\* as arrays.
    - Track patches at AOSC-Tracking/mesa @ aosc/mesa-25.2.0
    \(HEAD: 9443f6617f7118f5d16d6e42417b3f4ee1cabba9\).

Package(s) Affected
-------------------

- directx-headers: 1.618.2
- mesa: 1:25.2.5+dxheaders1.618.2
- mesa+32: 25.2.5
- fastfetch: 2.53.0-2

Security Update?
----------------

No

Build Order
-----------

```
#buildit mesa:+stage2 directx-headers mesa fastfetch
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`
- [x] LoongArch 64-bit (No SIMD) `loongarch64_nosimd`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

MingcongBai committed:
mesa: update to 25.2.5

- Revisit meson.options and enable options explicitly.
- Revise MESON_AFTER* as arrays.
- Enable all AArch64-specific drivers on amd64, helping with AArch64-x86
  emulation with sysroot-driven solutions (such as FEX).
- (loongson3) Disable rust features (rusticl, gfxstream Vulkan) due to
  libc (crate) brokenness.
- Reduce the feature set of stage 2 to simplify the logic.
- Remove or modify deprecated or removed meson flags in stage 2 to fix meson
  setup failure.
- Build with system directx-headers.
- Track patches at AOSC-Tracking/mesa @ aosc/mesa-25.2.5
  (HEAD: ff4dcf999bdd04254add94999c64a0d359d45de9).

Cyanoxygen commented:
OBS crashes upon starting a recording:

https://aosc.io/paste/detail?id=0e237f71-fad1-42b6-a1a0-7d17e3b5ffd5

MingcongBai committed:
mesa+32: update to 25.2.5+dxheaders1.618.2

- Enable all AArch64-specific drivers on amd64, helping with AArch64-x86
  emulation with sysroot-driven solutions (such as FEX).
- Disable Rust features due to 32-bit rustc 1.90.0 brokenness.
- Backport an upstream fix for graphical corruption on Intel cards.
- Restore DirectX-Headers.pc.
- We cannot use system directx-headers because the system one ships two
  64-bit static libraries.
- Track patches at AOSC-Tracking/mesa @ aosc/mesa-25.2.5
  (HEAD: ff4dcf999bdd04254add94999c64a0d359d45de9).

HmnSn committed:
fastfetch: rebuild to enable chafa and directx-headers

xry111 committed:
directx-headers: new, 1.618.2

Separated from Mesa so it can be used by other packages.

HmnSn commented:
fastfetch can detect gpu name on wsl2 now
![IMG_20251017_195050](https://github.com/user-attachments/assets/9b48f147-3b3b-45d6-9d62-e7d7f43660dc)
![IMG_20251017_195118](https://github.com/user-attachments/assets/db9f73a4-33c8-4ee6-9388-73e71fc82d82)

Cyanoxygen reviewed (approved):
### amd64

After resetting the configuration of the OBS Studio, it now records normally:
<img width="2007" height="2160" alt="image" src="https://github.com/user-attachments/assets/8051206b-3542-44a2-990e-4c5cfa3048b7" />

### Asahi

I tested this build with Asahi. With `DRM_ASAHI` enabled (in a recernt topic), the GPU now working flawlessly, I even played reVC.

Cyanoxygen commented:
> After resetting the configuration of the OBS Studio

Frankly, I didn't backup the configuration. Removing `~/.config/obs-studio`, and starting over just solves that issue.
